packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -480,6 +480,61 @@ const api = new apigateway.RestApi(this, 'books', {
   }
 })
 ```
+### Access Logging
+
+The following example will configure API Gateway to enable custom access logging.
+
+#### CloudWatch Logs
+
+```ts
+const logGroup = new cwlogs.LogGroup(this, "ApiGatewayAccessLogs");
+const api = new apigateway.RestApi(this, 'books', {
+  deployOptions: {
+    accessLogSetting: {
+        destinationArn: new apigateway.CloudWatchLogsDestination(logGroup),
+        format: JSON.stringify({
+            requestId: "$context.requestId",
+            ip: "$context.identity.sourceIp",
+            caller: "$context.identity.caller",
+            user: "$context.identity.user",
+            requestTime: "$context.requestTime",
+            httpMethod: "$context.httpMethod",
+            resourcePath: "$context.resourcePath",
+            status: "$context.status",
+            protocol: "$context.protocol",
+            responseLength: "$context.responseLength"
+        })
```

Comment:
We can do better here in terms of customer experience, and provide a more programming lang friendly and more a natural experience. Something like - 

```ts
format: {
  requestId: AuthorizerLogFormat.contextRequestId(),
  ip: AuthorizerLogFormat.contextIdentitySourceIp(),
  ...
}
```

or something along those lines.

The `JSON.stringify` can occur within the implementation of the construct, and hide it away from users.

Reply:
I totally agree with the rest of the comments in this revision, but this one is a bit controversial. 

Specifically, the format can be [43](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html#context-variable-reference) different context variables, those context variables can be arbitrary based on header names/parameter key names as well. The ordering of the keys is important in CLF/CSV format (also you can specify any format whatsoever, the above is just a clean json example). Due to it being an arbitrary string, that's specific to the customer's use case, I am unsure abstracting it away is a good idea, as it's dictating specific formats or ways.

Reply:
as an example, this is a valid accesslog format I made up on the fly that should be accepted:
`requestId[$context.requestId]` which would output: `requestId[21123-4145fds-342fds-5fdss]` in your access logs.

Reply:
> as an example, this is a valid accesslog format I made up on the fly that should be accepted:
> requestId[$context.requestId] which would output: requestId[21123-4145fds-342fds-5fdss] in your access logs.

If you used the code (I made up above), this would look like -

```ts
format: {
  requestId: `requestId[${AuthorizerLogFormat.contextRequestId()}]`
}
```

> Specifically, the format can be 43 different context variables, those context variables can be arbitrary based on header names/parameter key names as well.

I can see how bootstrapping 43 context variables and a bunch more stage variables, input variables, util variables can be hard. This is where we will have to get creative. We can start here -

```ts
AccessLogFormat.context('authorizer.principalId')
AccessLogFormat.input('params')
```

The ones that are most commonly used can then be added onto this and built up as necessary and over time - 

```
AccessLogFormat.CONTEXT_REQUEST_ID
// or
AccessLogFormat.contextRequestId()
// The latter probably since some of these can have additional suffixes like 'claims' and 'authorizer'
```